src/Analyzers/Core/Analyzers/NamingStyle/NamingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -80,6 +84,8 @@ void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
                     return;
                 }
 
+                RoslynDebug.AssertNotNull(syntaxContext.Compilation);
```

Comment:
📝 This appears to be a bug in the nullable annotations for the context type

Reply:
➡️ Submitted #47537 to fix this

Reply:
Thanks for looking at that. I wasn't sure about it because its a public API, so even though _semanticModel is non-nullable, you can't really trust it.

Reply:
#47537 is now merged, so this assertion is no longer necessary